hyperspy/io_plugins/impulse.py

```diff
@@ -0,0 +1,207 @@
+# ----------------------
+
+
+_logger = logging.getLogger(__name__)
+
+
+# At some point, if there is another readerw, whith also use csv file, it will
+# be necessary to mention the other reader in this message (and to add an
+# argument in the load function to specify the correct reader)
+invalid_file_error = (
+    "The csv reader can't import the file, please"
+    " make sure, that this is a valid Impulse log file."
+)
+invalid_filenaming_error = {
+    "The filename does not match Impulse naming, please"
+    " make sure that the filenames for the logfile and metadata file are unchanged."
+}
+no_metadata_file_error = {
+    "No _Metadata file detected in filepath, please"
+    " make sure that the Metadata file is included."
+}
+
+
+def file_reader(filename, *args, **kwds):
+    csv_file = impulseCSV(filename)
+    return _impulseCSV_log_reader(csv_file)
+
+
+def _impulseCSV_log_reader(csv_file):
+    csvs = []
+    for key in csv_file.logged_quantity_name_list:
+        try:
+            csvs.append(csv_file.get_dictionary(key))
+        except BaseException:
+            raise IOError(invalid_file_error)
+    return csvs
+
+
+class impulseCSV:
+    def __init__(self, filename):
+        self.filename = filename
+        self._parse_header()
+        self._read_data()
+
+    def _parse_header(self):
+        with open(self.filename, "r") as f:
+            s = f.readline()
+            self.column_names = s.strip().split(",")
+            if not self._is_impulse_csv_file():
+                raise IOError(invalid_file_error)
+            self._read_metadatafile()
+        self.logged_quantity_name_list = self.column_names[2:]
+
+    def _is_impulse_csv_file(self):
+        if "TimeStamp" in self.column_names and len(self.column_names) >= 3:
+            return True
+        else:
+            return False
+
+    def get_dictionary(self, quantity):
+        return {
+            "data": self._data_dictionary[quantity],
+            "axes": self._get_axes(),
+            "metadata": self._get_metadata(quantity),
+            "original_metadata": {"Impulse_header": self.original_metadata},
+        }
+
+    def _get_metadata(self, quantity):
+        return {
+            "General": {
+                "original_filename": os.path.split(self.filename)[1],
+                "title": "%s" % quantity,
+                "date": self.original_metadata["Experiment_date"],
+                "time": self.original_metadata["Experiment_time"],
+            },
+            "Signal": {
+                "signal_type": "",
+                "quantity": self._parse_quantity_units(quantity),
+            },
+        }
+
+    def _parse_quantity_units(self, quantity):
+        quantity_split = quantity.strip().split(" ")
+        if len(quantity_split) > 1:
+            if quantity_split[-1][0] == "(" and quantity_split[-1][-1] == ")":
+                return quantity_split[-1]
+            else:
+                return "NA"
+        else:
+            return "NA"
```

Comment:
`'NA'` can be ambiguous: better to return an empty string, or not to populate the metadata.